source/namedCurryN.js

```diff
@@ -0,0 +1,76 @@
+import curryN from './curryN.js';
+import curry from './curry.js';
+import fromPairs from './fromPairs.js';
+import toPairs from './toPairs.js';
+import _isPlaceholder from './internal/_isPlaceholder.js';
+import filter from './filter.js';
+import concat from './concat.js';
+import compose from './compose.js';
+import keys from './keys.js';
+
+/**
+ * Returns a curried equivalent of the provided function with named parameters.
+ * Has all the capabilities of curryN, but with named parameters.
+ * The following are equivalent, though returned order is not guaranteed:
+ *
+ *   - `g({ a: 1 })({ b: 2 })({ c: 3 })`
+ *   - `g({ a: 1 })({ b: 2, c: 3 })`
+ *   - `g({ b: 2, c: 3 })({ a: 1 })`
+ *   - `g({ c: 3, b: 2, a: 1 })`
+ *
+ * Secondly, the special placeholder value [`R.__`](#__) may be used to specify
+ * "gaps", allowing partial application of any combination of arguments,
+ * regardless of their positions. If `g` is as above and `_` is [`R.__`](#__),
+ * the following are equivalent:
+ *
+ *   - `g({ a: 3, b: 2, c: 1 })`
+ *   - `g({ a: 3, b: _, c: 1 })({ b: 2 })`
+ *   - `g({ a: _, b: _, c: _ })({ b: 2, a: 3 })({ c: 1 })`
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig Number -> (* -> a) -> (* -> a)
+ * @param {Number} length The arity for the returned function.
+ * @param {Function} fn The function to curry.
+ * @return {Function} A new, curried function.
+ * @see R.curryN, R.curry, R.partial
+ * @example
+ *
+ *      const addFourNumbers = ({ a, b, c, d }) => a + b + c + d;
+ *
+ *      const curriedAddFourNumbers = R.namedCurryN(4, addFourNumbers);
+ *      const f = curriedAddFourNumbers({ b: 1, c: _ });
+ *      const g = f({ a: 3 });
+ *      g({ d: 4, c: 2 }); //=> 10
+ */
+
+function filterPlaceholders(mappedPairs) {
+  return filter(function([, val]) {
+    return !_isPlaceholder(val);
+  }, mappedPairs);
+}
+
+function prepareEntries(acc, mappedPairs) {
+  return compose(curry(concat)(acc), filterPlaceholders)(mappedPairs);
+}
```

Comment:
This feels especially strange for a Ramda implementation.  Besides the fact that Ramda uses internal `curry2`, `curry3`, etc. helpers, I can see no reason to write `compose (/* fns */) (/* args */)`.  Here I see no advantage to this over

```js
  return concat(acc, filterPlaceholders (mappedPairs));
```

or something similar with the native `concat`

Reply:
Fair enough, I had native `concat` originally but thought I'd try point-free. Though I think in my refactor I got lost a little in the abstraction, and you're right, I don't think it's necessary. I'll fix.

Reply:
Before spending time on this particular fix, though, do note the main issue above.  It seems you're matching on the wrong things.